ts/src/base/Exchange.ts

```diff
@@ -4212,22 +4242,18 @@ export default class Exchange {
                     }
                 }
             } else {
-                // if currencyCode was provided, then we try to find if that currencyCode has a replacement (i.e. ERC20 for ETH) or is in the currency
-                const defaultNetworkCodeReplacements = this.safeValue (this.options, 'defaultNetworkCodeReplacements', {});
-                if (currencyCode in defaultNetworkCodeReplacements) {
-                    // if there is a replacement for the passed networkCode, then we use it to find network-id in `options->networks` object
-                    const replacementObject = defaultNetworkCodeReplacements[currencyCode]; // i.e. { 'ERC20': 'ETH' }
-                    const keys = Object.keys (replacementObject);
-                    for (let i = 0; i < keys.length; i++) {
-                        const key = keys[i];
-                        const value = replacementObject[key];
-                        // if value matches to provided unified networkCode, then we use it's key to find network-id in `options->networks` object
-                        if (value === networkCode) {
-                            networkId = this.safeString (networkIdsByCodes, key);
-                            break;
-                        }
+                // check if user incorrectly passed mainnet-vs-protocol networkCode, eg:
+                // - for ETH coin passed `ERC20` networkCode
+                // - for USDT coin passed `ETH` networkCode
+                const chainDescriptors = this.safeList (this.options, 'chainDescriptors', []);
+                for (let i = 0; i < chainDescriptors.length; i++) {
+                    const entry = chainDescriptors[i];
+                    if (networkCode === entry['secondary'] || networkCode === entry['primary']) {
+                        networkId = this.safeString2 (networkIdsByCodes, entry['primary'], entry['secondary']);
                     }
-                } else {
+                }
+                // if networkId wasn't found yet, then check inside currency.networks
+                if (networkId === undefined) {
```

Comment:
this change happens in `_code_to_id` and only in "else" clause when currencyCode is provided.